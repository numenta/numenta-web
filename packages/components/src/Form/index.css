/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.form {
  composes: pa0 from 'tachyons';
  composes: mb3 from 'tachyons';
  composes: mh0 from 'tachyons';
  composes: mt0 from 'tachyons';
}
