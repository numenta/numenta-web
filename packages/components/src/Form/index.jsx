// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


const Form = ({action, children, name, onSubmit}) => (
  <form
    acceptCharset="UTF-8"
    action={action}
    autoComplete="off"
    className={styles.form}
    encType="multipart/form-data"
    id={name}
    method="post"
    name={name}
    onSubmit={onSubmit}
  >
    {children}
  </form>
)

Form.propTypes = {
  action: React.PropTypes.string,
  children: React.PropTypes.node.isRequired,
  name: React.PropTypes.string,
  onSubmit: React.PropTypes.func,
}

export default Form
