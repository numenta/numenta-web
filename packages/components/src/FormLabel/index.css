/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.formLabel {
  composes: f5 from 'tachyons';
  composes: fl from 'tachyons';
  composes: fw6 from 'tachyons';
  composes: mb3 from 'tachyons';
  composes: mr3 from 'tachyons';
  composes: relative from 'tachyons';
  composes: v-top from 'tachyons';
  composes: w4 from 'tachyons';
  top: 0.5rem;
}
