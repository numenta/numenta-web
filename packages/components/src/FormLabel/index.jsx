// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


const FormLabel = ({children, htmlFor}) => (
  <label className={styles.formLabel} htmlFor={htmlFor}>
    {children}
  </label>
)

FormLabel.propTypes = {
  children: React.PropTypes.node.isRequired,
  htmlFor: React.PropTypes.string,
}

export default FormLabel
