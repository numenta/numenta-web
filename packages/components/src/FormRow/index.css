/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.formRow {
  composes: cb from 'tachyons';
  composes: cf from 'tachyons';
  composes: flex from 'tachyons';
  composes: mb2 from 'tachyons';
  composes: mr4 from 'tachyons';
  flex-direction: row;
}
