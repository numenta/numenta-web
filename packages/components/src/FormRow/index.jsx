// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


const FormRow = ({children}) => (
  <div className={styles.formRow}>
    {children}
  </div>
)

FormRow.propTypes = {
  children: React.PropTypes.node.isRequired,
}

export default FormRow
