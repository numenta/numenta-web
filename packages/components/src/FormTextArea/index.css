/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._focus:focus {
  border: 2px solid #1aa0db;
  color: #000;
  outline: 0;
}


.formTextArea {
  composes: ba from 'tachyons';
  composes: b--black-20 from 'tachyons';
  composes: br2 from 'tachyons';
  composes: f5 from 'tachyons';
  composes: flex-auto from 'tachyons';
  composes: fw3 from 'tachyons';
  composes: h4 from 'tachyons';
  composes: input-reset from 'tachyons';
  composes: pb1 from 'tachyons';
  composes: ph2 from 'tachyons';
  composes: pt2 from 'tachyons';
  composes: silver from 'tachyons';
  composes: _focus;
}
