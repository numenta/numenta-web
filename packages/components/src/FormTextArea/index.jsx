// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


const FormTextArea = ({children, name, placeholder}) => (
  <textarea
    className={styles.formTextArea}
    id={name}
    name={name}
    placeholder={placeholder}
  >
    {children}
  </textarea>
)

FormTextArea.propTypes = {
  children: React.PropTypes.node,
  name: React.PropTypes.string.isRequired,
  placeholder: React.PropTypes.string.isRequired,
}

export default FormTextArea
