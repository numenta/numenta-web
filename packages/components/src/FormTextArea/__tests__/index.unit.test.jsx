import React from 'react'
import renderer from 'react-test-renderer'

import Form from '../../Form'
import FormLabel from '../../FormLabel'
import FormRow from '../../FormRow'
import FormTextArea from '../../FormTextArea'


describe('FormTextArea React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Form
        action="http://numenta.com"
        name="form"
        onSubmit={() => {}}
      >
        <FormRow>
          <FormLabel for="field">
            Label
          </FormLabel>
          <FormTextArea name="field" placeholder="Default">
            Contents
          </FormTextArea>
        </FormRow>
      </Form>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
