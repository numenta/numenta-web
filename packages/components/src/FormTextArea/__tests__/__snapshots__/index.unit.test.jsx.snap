exports[`FormTextArea React component Renders correctly 1`] = `
<form
  acceptCharset="UTF-8"
  action="http://numenta.com"
  autoComplete="off"
  className="form"
  encType="multipart/form-data"
  id="form"
  method="post"
  name="form"
  onSubmit={[Function]}>
  <div
    className="formRow">
    <label
      className="formLabel"
      htmlFor={undefined}>
      Label
    </label>
    <textarea
      className="formTextArea"
      id="field"
      name="field"
      placeholder="Default">
      Contents
    </textarea>
  </div>
</form>
`;
